core/src/main/java/org/neo4j/graphalgo/core/neo4jview/RelationIteratorImpl.java

```diff
@@ -1,27 +1,71 @@
 package org.neo4j.graphalgo.core.neo4jview;
 
 import org.neo4j.cursor.Cursor;
+import org.neo4j.graphalgo.api.Graph;
 import org.neo4j.graphalgo.api.RelationCursor;
+import org.neo4j.graphdb.Direction;
+import org.neo4j.graphdb.Transaction;
 import org.neo4j.kernel.api.ReadOperations;
+import org.neo4j.kernel.api.Statement;
+import org.neo4j.kernel.api.exceptions.EntityNotFoundException;
 import org.neo4j.kernel.impl.api.store.RelationshipIterator;
+import org.neo4j.kernel.impl.core.ThreadToStatementContextBridge;
+import org.neo4j.kernel.internal.GraphDatabaseAPI;
 import org.neo4j.storageengine.api.RelationshipItem;
 
-class RelationIteratorImpl extends AbstractRelationIterator<RelationCursor> {
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Iterator;
 
-    private final RelationCursor cursor = new RelationCursor();
+class RelationIteratorImpl implements Iterator<RelationCursor>, Closeable {
 
-    public RelationIteratorImpl(long nodeId, ReadOperations read, RelationshipIterator iterator) {
-        super(nodeId, read, iterator);
-        cursor.sourceNodeId = (int) nodeId;
+    private final Graph graph;
+    private final GraphDatabaseAPI api;
+    private final Transaction transaction;
+    private final Statement statement;
+    private final RelationshipIterator iterator;
+    private final ReadOperations read;
+    private final RelationCursor cursor;
+    private final long originalNodeId;
+
+    RelationIteratorImpl(Graph graph, GraphDatabaseAPI api, int sourceNodeId, Direction direction, int relationTypeId) throws EntityNotFoundException {
+        this.graph = graph;
+        this.api = api;
+        transaction = api.beginTx();
+        statement = api.getDependencyResolver()
+                .resolveDependency(ThreadToStatementContextBridge.class)
+                .get();
```

Comment:
We think it is very uncommon to use one iterator from several threads. I agree with you to document this.